core/domain/platform_feature_services.py

```diff
@@ -198,11 +216,11 @@ def _evaluate_feature_flag_values_for_context(feature_names_set, context):
     for feature_name in feature_names_set:
         param = registry.Registry.get_platform_parameter(
             feature_name)
-        result_dict[feature_name] = param.evaluate(context)
+        result_dict[feature_name] = bool(param.evaluate(context))
```

Comment:
Why does it need to be a boolean? I am not sure this is true in general... please check with @BenHenning on this.

Ditto elsewhere.

Reply:
I think not every value is going to return true, because there is a test to check the False case. https://github.com/oppia/oppia/blob/f8b0f246dd58d07ea38dedbfe42ee584cee15f3b/core/domain/platform_feature_services_test.py#L163

Reply:
No, I mean, I'm not sure it's true that every value is going to be a boolean. Is it possible for it to be a string etc.?

Reply:
I think if we look at the docstring, it says `the values are boolean results of corresponding flags.` So I assumed that the values is going to be boolean converted values. 
https://github.com/oppia/oppia/blob/1bd5ab86be391bb9ee896cfe0514ab3ab8093665/core/domain/platform_feature_services.py#L185

@BenHenning it would be very helpful if you clear the ambiguity :)

Reply:
OK, I think that seems reasonable. But then why do you need to cast to a boolean here? Shouldn't "evaluate" return a boolean result?

Reply:
It's not necessary that `evaluate` is always going to return a `boolean` value. because if look at the `evaluate` function's code then we can see that it can also return `default_value` ( which can be a string and int as well ).

https://github.com/oppia/oppia/blob/c5c4515b361f0b21494ab4f7333ef9610665060c/core/domain/platform_parameter_domain.py#L847

Also, from test cases, It is clearer that not only bool is returned https://github.com/oppia/oppia/blob/c5c4515b361f0b21494ab4f7333ef9610665060c/core/domain/platform_parameter_domain_test.py#L1969
https://github.com/oppia/oppia/blob/c5c4515b361f0b21494ab4f7333ef9610665060c/core/domain/platform_parameter_domain_test.py#L2006

Reply:
Please correct me if I'm wrong somewhere!

Reply:
OK, so then maybe you need to check this with @BenHenning (maybe send an email).  Because it seems like the docstring is saying one thing, and the signature for evaluate() is telling you a different thing.

Reply:
Email is sent!! and waiting for response. In mean while can you please take a look at other comments?

Reply:
assertion is added!! PTAL!

Reply:
@seanlip